coveo-functools/coveo_functools/flex/deserializer.py

```diff
@@ -61,23 +62,25 @@ def convert_kwargs_for_unpacking(dirty_kwargs: Dict[str, Any], *, hint: MetaHint
 
 
 @overload
-def deserialize(value: Any, *, hint: Type[T]) -> T:
+def deserialize(value: Any, *, hint: Type[T], strict: bool = False) -> T:
     ...
 
 
 @overload
-def deserialize(value: Any, *, hint: T) -> T:
+def deserialize(value: Any, *, hint: T, strict: bool = False) -> T:
     """This overload tricks mypy in passing typing annotations as types, such as List[str]."""
 
 
-def deserialize(value: Any, *, hint: Union[T, Type[T]]) -> T:
+def deserialize(value: Any, *, hint: Union[T, Type[T]], strict: bool = False) -> T:

```

Comment:
Maybe this could be called `raise` or something similar that's not a reserved keyword (python might not like you using `raise`). This would make the usage pretty obvious to people who have never seen the option or are not familiar with how `flex` outputs values.